pkg/gofr/handler.go

```diff
@@ -87,8 +87,18 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	select {
 	case <-c.Context.Done():
-		// If the context's deadline has been exceeded, return a timeout error response
-		if errors.Is(c.Err(), context.DeadlineExceeded) {
+		// Handle different context cancellation scenarios
+		ctxErr := c.Context.Err()
+
+		switch {
+		case errors.Is(ctxErr, context.Canceled):
+			// Client canceled the request (e.g., closed browser tab)
+			err = gofrHTTP.ErrorClientClosedRequest{}
+		case errors.Is(ctxErr, context.DeadlineExceeded):
+			// Server-side timeout occurred
+			err = gofrHTTP.ErrorRequestTimeout{}
+		default:
+			// Fallback for other context errors
 			err = gofrHTTP.ErrorRequestTimeout{}
 		}
```

Comment:
I'm unsure about the need of a case when the content of branches are identical


```suggestion
		default:
			// Fallback for other context errors
 			err = gofrHTTP.ErrorRequestTimeout{}
 		}
```

Unless you want to return another error when not canceled or timeout